src/connect.rs

```diff
@@ -72,15 +72,17 @@ pub fn connect_to(peer_info: CrustInfo, send_after_connect: Option<WireMsg>) ->
 
 fn handle_new_connection_res(
     peer_addr: SocketAddr,
-    new_peer_conn_res: Result<quinn::NewClientConnection, quinn::ConnectionError>,
+    new_peer_conn_res: Result<quinn::NewConnection, quinn::ConnectionError>,
 ) {
     let new_peer_conn = match new_peer_conn_res {
         Ok(new_peer_conn) => new_peer_conn,
         Err(e) => return handle_connect_err(peer_addr, &From::from(e)),
     };
+    current_thread::spawn(new_peer_conn.driver.map_err(|_| ()));
 
     println!("Successfully connected to peer: {}", peer_addr);
 
+    let q_conn = new_peer_conn.connection;
```

Comment:
Why is this diff necessary ?

Reply:
Otherwise `new_peer_conn` would be used inside `ctx_mut({})` context which is impossible, because `current_thread::spawn(new_peer_conn.driver.map_err(|_| ()));` moves `driver` from `new_peer_conn`